docs/schema-support.md

```diff
@@ -0,0 +1,72 @@
+
+In order to diagnose these issues, you first need to make sure that those policies aren’t in the schema. Download and search through the [sam schema](https://github.com/awslabs/goformation/blob/master/schema/sam.schema.json) and look for “**AWS::Serverless::Function.SAMPolicyTemplate**”. Under the “properties” field try and find whatever policy is missing.
+
+-   If the policy is missing then clone the [goformation](https://github.com/awslabs/goformation) project, add the policy under “**AWS::Serverless::Function.SAMPolicyTemplate**” in [sam-2016-10-31.json](https://github.com/awslabs/goformation/blob/master/generate/sam-2016-10-31.json) and run `go generate` at the root of the repository. This will generate the sam.schema.json that we pull into aws-toolkit-vscode. See PRs: [#450](https://github.com/awslabs/goformation/pull/450), [#448](https://github.com/awslabs/goformation/pull/448), [#449](https://github.com/awslabs/goformation/pull/449) for examples
+-   If the policy is not missing then verify that the reference ($ref) it points to in the JSON schema is whatever type the AWS documentation says.
+    -   E.g. If a user is having problems with LambdaInvokePolicy
+        -   Find the AWS documentation for that policy: https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-policy-template-list.html#lambda-invoke-policy
+        -   Verify that the schema has the same "Resource" properties as the documentation
+            -   If the "Resource" properties are different, you might need to add a new field to the JSON schema according to the specs of the AWS documentation and have the policy point to that in the JSON Schema. Similiar to [this change](https://github.com/awslabs/goformation/pull/449/files#diff-f6615b52e8a9fb8465ba150df1fdba3ce3ce7a262a43fa2e5c5f0d4057c09456R1669).
+            -   If the "Resource" properties are the same as the documentation then it should be validating correctly. If you are absolutely sure that the YAML is correct according to the documentation and the schema isn't the issue then raise an issue on [VSCode-YAML](https://github.com/redhat-developer/vscode-yaml)
+
+### Incorrect type
+
+Ocassionally we will get incorrect type issues like:
+
+```
+Incorrect type. Expected "string".yaml-schema: file://.../globalStorage/amazonwebservices.aws-toolkit-vscode/sam.schema.json
+```
+
+when the YAML looks like it's correct. These issues are hard to diagnose because it means the yaml that was entered isn’t what vscode-yaml/yaml-language-server expects. This can happen for multiple reasons.
+
+1. The yaml is syntatically valid but not semantically valid (according to the schema)
+
+    - In this case, visit the AWS documentation and make sure what was entered makes sense (type wise) according to the documentation.
```

Comment:
Maybe we can add this link https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-reference.html ?